plugins/docker/personal_access_token.go

```diff
@@ -0,0 +1,68 @@
+package docker
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func PersonalAccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.PersonalAccessToken,
+		DocsURL:       sdk.URL("https://docs.docker.com/docker-hub/access-tokens/"),
+		ManagementURL: sdk.URL("https://docs.docker.com/docker-hub/access-tokens/"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Docker.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 18,
+					Charset: schema.Charset{
+						Lowercase: true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryDockerConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"DOCKERHUB_TOKEN": fieldname.Token, // TODO: Check if this is correct
```

Comment:
@roy9495 Are you sure `DOCKERHUB_TOKEN` exists and is interpreted by Docker CLI? I couldn't get this to work

Reply:
No sir, I am not sure but [here](https://docs.docker.com/build/ci/github-actions/) it was mentioned. So, I used it. I could not set it up on my PC, so I set it up on Gitpod.